src/codegen/mod.rs

```diff
@@ -3948,11 +3978,28 @@ pub(crate) fn codegen(
             &(),
         );
 
+        if context.options().dynamic_library_name.is_some() {
```

Comment:
nit: `if let Some(ref lib_name) = context.options().dynamic_library_name {`.

Then use that instead of unwrapping around.